packages/salesforce-adapter/src/elements_url_retreiver/lightining_url_resolvers.ts

```diff
@@ -0,0 +1,174 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { isObjectType, Element, isField, Field, isInstanceElement, isType, ElementIDResolver, CORE_ANNOTATIONS, isElement, isReferenceExpression } from '@salto-io/adapter-api'
+import { getParents } from '@salto-io/adapter-utils'
+import { metadataType } from '../transformers/transformer'
+import { getInternalId } from '../filters/utils'
+
+export type UrlResolver = (element: Element, baseUrl: URL, _elementIDResolver: ElementIDResolver) =>
+  Promise<URL | undefined>
+
+const GENERAL_URLS_MAP: Record<string, string> = {
+  PermissionSet: 'lightning/setup/PermSets/home',
+  PermissionSetGroup: 'lightning/setup/PermSetGroups/home',
+  ApexClass: 'lightning/setup/ApexClasses/home',
+  Flow: 'lightning/setup/Flows/home',
+  Profile: 'lightning/setup/EnhancedProfiles/home',
+  CustomMetadata: 'lightning/setup/CustomMetadata/home',
+  CustomPermission: 'lightning/setup/CustomPermissions/home',
+  ApexComponent: 'lightning/setup/ApexComponents/home',
+  ApexPage: 'lightning/setup/ApexPages/home',
+  EmailTemplate: 'lightning/setup/CommunicationTemplatesEmail/home',
+  ReportType: 'lightning/setup/CustomReportTypes/home',
+  AnalyticSnapshot: 'lightning/setup/AnalyticSnapshots/home',
+  ApexTrigger: 'lightning/setup/ApexTriggers/home',
+  Queue: 'lightning/setup/Queues/home',
+  LightningComponentBundle: 'lightning/setup/LightningComponentBundles/home',
+  Report: 'lightning/o/Report/home',
+  Dashboard: 'lightning/o/Dashboard/home',
+}
+
+const SETTINGS_URLS_MAP: Record<string, string> = {
+  BusinessHoursSettings: 'lightning/setup/BusinessHours/home',
+  LanguageSettings: 'lightning/setup/LanguageSettings/home',
+  MyDomainSettings: 'lightning/setup/OrgDomain/home',
+  ApexSettings: 'lightning/setup/ApexSettings/home',
+}
+
+const getTypeIdentifier = (element: Element | undefined): string | undefined =>
+  (element === undefined ? undefined : (getInternalId(element) ?? element.annotations.apiName))
+
+const getFieldIdentifier = (element: Field): string =>
+  (getInternalId(element) ?? element.annotations.relationshipName ?? element.name)
+
+const genernalConstantsResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  if (isObjectType(element) && metadataType(element) in GENERAL_URLS_MAP) {
+    return new URL(`${baseUrl}${GENERAL_URLS_MAP[element.annotations.metadataType]}`)
+  }
+  return undefined
+}
+
+const settingsConstantsResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  if (metadataType(element) in SETTINGS_URLS_MAP) {
+    return new URL(`${baseUrl}${SETTINGS_URLS_MAP[metadataType(element)]}`)
+  }
+  return undefined
+}
+
+const assignmentRulesResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
```

Comment:
nit: in javascript / typescript parameters can be omitted, if you are not using the last one, just define the function without it
```suggestion
const assignmentRulesResolver: UrlResolver = async (element, baseUrl) => {
```